src/geogenalg/cluster.py

```diff
@@ -0,0 +1,40 @@
+#  Copyright (c) 2025 National Land Survey of Finland (Maanmittauslaitos)
+#
+#  This file is part of geogen-algorithms.
+#
+#  This source code is licensed under the MIT license found in the
+#  LICENSE file in the root directory of this source tree.
+
+import geopandas as gpd
+
+
+def reduce_nearby_points(
+    input_gdf: gpd.GeoDataFrame, reduce_threshold: float
+) -> gpd.GeoDataFrame:
+    """Reduce the number of points by clustering and replacing them with their centroid.
+
+    Args:
+    ----
+        input_gdf: Input GeoDataFrame with point geometries
+        reduce_threshold: Distance used for buffering and clustering
+
+    Returns:
+    -------
+        A GeoDataFrame containing centroids of clusters.
```

Comment:
What happens to the attributes of the clustered (merged) points? For example if three nearby points enter and they have different identifiers and perhaps other attributes and the output is only one point, what are the attributes of that point?

If the attributes of the clustered points are chosen "randomly" or they don't matter at the moment and the issue will be resolved later, then you could indicate that in the docstring (and perhaps add a comment to the code) so the user of the function is aware of this. We can discuss this in more detail in Teams if needed.